lua/toggleterm.lua

```diff
@@ -24,6 +24,17 @@ local preferences = {
 -- State
 -----------------------------------------------------------
 local terminals = {}
+local persistent = {}
+
+--- Get the size of the split. Order of priority is as follows:
+---   1. The size argument is a valid number > 0
+---   2. There is persistent width/height information from prev open state
+---   3. Default/base case perference size
+--- @param size number
+local function get_size(size)
+  local persist_size = preferences.direction == "horizontal" and persistent['height'] or persistent['width']
+  return (size and size > 0) and size or persist_size and persist_size or preferences.size
```

Comment:
I don't think the `persist_size and persist_size` duplication is necessary. Like `size or persist_size or preferences.size` haven't actually run it so don't know that I'm not missing something